Core/include/Acts/Detector/IndexedGridFiller.hpp

```diff
@@ -0,0 +1,385 @@
+/// indexed grid.
+///
+/// The grid filling then completes the empty bins in between and
+/// expands if necessary.
+struct PolyhedronReferenceGenerator {
+  /// Also use the barycenter
+  bool addBarycenter = true;
+
+  /// The number of segments to approximate (1 means extrema points only)
+  unsigned int nSegments = 1;
+
+  /// Helper to access the Center point of for filling the grid
+  ///
+  /// @param gctx the geometry context of this operation
+  /// @param surface the surface for which the reference point is to be accessed
+  ///
+  /// @return a vector of referene points for filling
+  const std::vector<Vector3> references(const GeometryContext& gctx,
+                                        const Surface& surface) const {
+    // Create the return  vector
+    std::vector<Vector3> rPositions;
+    auto pHedron = surface.polyhedronRepresentation(gctx, nSegments);
+    rPositions.insert(rPositions.end(), pHedron.vertices.begin(),
+                      pHedron.vertices.end());
+    // Add the barycenter if configured
+    if (addBarycenter) {
+      Vector3 bc(0., 0., 0.);
+      std::for_each(rPositions.begin(), rPositions.end(),
+                    [&](const auto& p) { bc += p; });
+      bc *= 1. / rPositions.size();
+      rPositions.push_back(bc);
+    }
+    return rPositions;
+  }
+};
+
+/// A helper class that fills surfaces into predefined grids
+struct IndexedGridFiller {
+  // Bin expansion where needed
+  std::vector<std::size_t> binExpansion = {};
+
+  // An ouput logging level
+  Logging::Level logLevel = Logging::INFO;
+
+  /// @brief This method takes a collection of objects and fills them
+  /// into an index grid - it uses a reference generator for grid query points
+  /// and then completes the bins in between.
+  ///
+  /// It also allows for expanding the fill view.
+  ///
+  /// @tparam index_grid the type of the index grid
+  /// @tparam indexed_objects the type of the object container
+  /// @tparam reference_generator the generator for reference points to be filled
+  ///
+  /// @param gctx the geometry context of the operation
+  /// @param iGrid [in,out] the index grid object to be filled
+  /// @param iObjects the object container to be indexed
+  /// @param rGenerator the reference point generator for position queries
+  /// @param aToAll the indices that are assigned to all bins
+  ///
+  /// @note as this is a Detector module, the objects within the indexed_objects container
+  /// are assumed to have pointer semantics
+  ///
+  template <typename index_grid, typename indexed_objects,
+            typename reference_generator>
+  void fill(
+      const GeometryContext& gctx, index_grid& iGrid,
+      const indexed_objects& iObjects, const reference_generator& rGenerator,
+      const typename index_grid::grid_type::value_type& aToAll = {}) const {
+    // Screen output
+    ACTS_LOCAL_LOGGER(getDefaultLogger("IndexedGridFiller", logLevel));
```

Comment:
sure, will do.

Reply:
@paulgessinger - do you have an example, how you would like to do it that we have a consistent handling of the output logger, I know you worked on this lately, so I will just follow your pattern, naming.

Reply:
Thinking about it - is this really necessary? These classes are only called once during construction of the geometry, and the log level can already be set through the class configuration - what would be the benefit of changing it to a function argument?

Reply:
Yes, since this logger always writes to `cout`, so cannot integrate with the e.g. Athena logging infrastructure at all.

You could add

```cpp
const Logger& logger = getDefaultLogger("IndexedGridFiller", Logging::INFO)
```

to the function signature as the last argument to get basically what you have now. Ideally though you would derive this logger from the top component down, thereby passing through the log level.

Overall, I'm trying to get rid of `ACTS_LOCAL_LOGGER` to discourage this pattern (I already had to do quite a bit of cleanup).

Reply:
Changed to private member.